entity-framework/core/what-is-new/ef-core-9.0/whatsnew.md

```diff
@@ -198,153 +139,176 @@ info: 6/10/2024 19:06:00.017 CosmosEventId.ExecutingSqlQuery[30100] (Microsoft.E
+  * `string.Contains`, `StartsWith` and `EndsWith` now support `StringComparison.OrdinalIgnoreCase`.
 
-EF Core still extracts the partition key values when executing this query:
+For the full list of querying improvements, see [this issue](https://github.com/dotnet/efcore/issues/33033):
 
-```output
-info: 6/10/2024 19:24:46.581 CosmosEventId.ExecutingSqlQuery[30100] (Microsoft.EntityFrameworkCore.Database.Command)
-      Executing SQL query for container 'UserSessionContext' in partition '["Microsoft","99a410d7-e467-4cc5-92de-148f3fc53f4c"]' [Parameters=[]]
-      SELECT c
-      FROM root c
-      WHERE ((c["Discriminator"] = "UserSession") AND CONTAINS(c["Username"], "a"))
+### Improved modeling aligned to Cosmos and JSON standards
+
+EF 9.0 maps to Cosmos DB documents in ways which are more natural for a JSON-based document database, and help interoperate with other systems accessing your documents. Although this entails breaking changes, APIs exist which allow reverting back to the pre-9.0 behavior in all cases.
+
+#### Simplified `id` properties without discriminators
+
+First, previous versions of EF inserted the discriminator value into the JSON `id` property, producing documents such as the following:
+
+```json
+{
+    "id": "Blog|1099",
+    ...
+}
 ```
 
-This query does not include the `SessionId`, so it cannot target a single partition. However, it will still be a targeted, cross-partition query returning data for all sessions of a single tenant and user ID.
+This was done in order to allow for documents of different types (e.g. Blog and Post) and the same key value (1099) to exist within the same container partition. Starting with EF 9.0, the `id` property contains contains only the key value:
 
-Likewise, if only the top value in the hierarchy is specified, then it will be used on its own. For example:
+```json
+{
+    "id": 1099,
+    ...
+}
+```
 
-<!--
-            var tenantId = "Microsoft";
+This is a more natural way to map to JSON, and makes it easier for external tools and systems to interact with EF-generated JSON documents; such external systems aren't generally aware of the EF discriminator values, which are by default derived from .NET types.
 
-            var sessions = await context.Sessions
-                .Where(
-                    e => e.TenantId == tenantId
-                         && e.Username.Contains("a"))
-                .ToListAsync();
--->
-[!code-csharp[TopOnePartitionKey](../../../../samples/core/Miscellaneous/NewInEFCore9.Cosmos/HierarchicalPartitionKeysSample.cs?name=TopOnePartitionKey)]
+Note this is a breaking change, since EF will no longer be able to query existing documents with the old `id` format. An API has been introduced to revert to the previous behavior, see the [breaking change note](xref:core/what-is-new/ef-core-9.0/breaking-changes#cosmos-id-property-changes) and the [the documentation](xref:core/providers/cosmos/index#Discriminators) for more details.
```

Comment:
```suggestion
Note this is a breaking change, since EF will no longer be able to query existing documents with the old `id` format. An API has been introduced to revert to the previous behavior, see the [breaking change note](xref:core/what-is-new/ef-core-9.0/breaking-changes#cosmos-id-property-changes) and the [the documentation](xref:core/providers/cosmos/modeling#Discriminators) for more details.
```